students/MicahBraun/Lesson 04/mailroom_2.py

```diff
@@ -0,0 +1,287 @@
+        print(optA + optB.center(37, ' ') + optC.center(36, ' ') + optD.center(25, ' ') + '\n')
+
+        switch_function = {
+            'A': thankyou,  # dictionary holding menu options
+            'B': createreport,
+            'C': writeletters,
+            'D': quit_program,
+        }
+        usr_sel = (input("Menu Selection: ")).upper()  # user input for menu
+        if usr_sel in switch_function:  # if check (used to display error msg)
+            switch_function.get(usr_sel)()  # get user entry from dict
+        else:
+            print("Invalid Entry. Only enter A, B, C, or D.")  # display error on any other entry
+        continue  # continue back to start of menu
+
+
+def thankyou():
+    """Function enables user to add to dictionary data (adding new donors and dontation amounts as well as
+    number of donations and average donations for new keys, and also updates existing keys."""
+    print("Leaving menu...\n")
+    while True:
+        print("Enter the name of the person you are writing to (or enter 'list' to see a list of names or Q to quit) ")
+        fname_prompt = input("First Name: ").strip().capitalize()  # first name variable (strip any spaces)
+        if fname_prompt.upper() == "Q":  # if Q, then quit to menu()
+            menu()
+        elif fname_prompt.lower() == "list":
+            displaylist()
+        else:
+            lname_prompt = input("Last Name: ").strip().capitalize()  # last name variable
+            if lname_prompt.upper() == "Q":  # if Q, back to menu
+                menu()
+            elif lname_prompt.lower() == "list":  # if list, then list display
+                displaylist()
+            else:
+                key = fname_prompt + " " + lname_prompt
+                if key in donor_dict.keys():
+                    existing_donor = input(
+                        "That value is already in the list! Do you want to proceed with that selection? (Y/N): ")
+                    if existing_donor.upper() == "Y":  # if user proceeds, print display
+                        donation_amt = int(input("Enter in the donation amount from Donor {0}: $".format(key)))
+                        if str(donation_amt).upper() == "Q":
+                            menu()
+                        else:
+                            print(
+                                '{0} has donated ${1:,.2f}'.format(key, donation_amt))  # display name and donation amt
+                            int(donation_amt)
+                            get_index = 0  # index variable
+                            for item in range(0, len(donor_dict)):  # set index var to current name
+                                if list(donor_dict.values())[item][0] == key:
+                                    get_index = item
+                                    break
+                            firstname = list(donor_dict.values())[get_index][0].split(' ', ).pop(
+                                0)  # separate first name, create variable
```

Comment:
You already have this value stored in `fname_prompt`, no need to reassign it here in this way